src/Controller/Project/CcaController.php

```diff
@@ -0,0 +1,205 @@
+use Symfony\Component\Form\Extension\Core\Type\HiddenType;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+
+class CcaController extends AbstractController
+{
+    /**
+     * @Security("has_role('ROLE_SUIVEUR')")
+     * @Route("/suivi/cca", name="project_cca_index")
+     *
+     * @return RedirectResponse|Response
+     */
+    public function index(): Response
+    {
+        $em = $this->getDoctrine()->getManager();
+
+        $ccas = $em->getRepository(Cca::class)->findAll();
+        // On veut récupérer leur nombre pour l'afficher entre parenthèses
+        $bdcs = $em->getRepository(Ce::class)->findBy(['type' => Ce::TYPE_BDC]);
+
+        $nbBdc = [];
+        foreach ($ccas as $cca) {
+            $nb = 0;
+            foreach ($bdcs as $bdc) {
+                if ($bdc->getEtude()->getCca() === $cca) {
+                    ++$nb;
+                    // L'efficacité de cette méthode étant terrible (peut-être cela pouvait-il se résoudre avec de meilleures requêtes SQL...) on l'améliore un peu en réduisant la taille de l'array $bdcs
+                    $key = array_search($bdc, $bdcs);
+                    unset($bdcs[$key]);
+                }
+            }
+            $nbBdc[] = $nb;
+        }
+        return $this->render('Project/Cca/index.html.twig', [
+            'controller_name' => 'CcaController',
+            'ccas' => $ccas,
+            'nbBdc' => $nbBdc,
+        ]);
+    }
+
+    /**
+     * @Security("has_role('ROLE_SUIVEUR')")
+     * @Route("/suivi/cca/voir/{id}", name="project_cca_voir")
+     *
+     * @return RedirectResponse|Response
+     */
+    public function voir(Cca $cca): Response
+    {
+        return $this->render('Project/Cca/voir.html.twig', [
+            'controller_name' => 'CcaController',
+            'bdcs' => $this->getBdcs($cca),
+            'cca' => $cca,
+        ]);
+    }
+
+    /**
+     * @Security("has_role('ROLE_SUIVEUR')")
+     * @Route("/suivi/cca/bdc/{id}", name="project_cca_bdc")
+     *
+     * @return RedirectResponse|Response
+     */
+    public function voirBdc(Cca $cca): Response
+    {
+        return $this->render('Project/Cca/bdcs.html.twig', [
+            'controller_name' => 'CcaController',
+            'cca' => $cca,
+            'bdcs' => $this->getBdcs($cca),
+        ]);
+    }
+
+    private function getBdcs(Cca $cca)
+    {
+        return $this->getDoctrine()->getManager()->getRepository(Ce::class)->findAllBdcByCca($cca);
+    }
+
+    /**
+     * @Security("has_role('ROLE_CA')")
+     * @Route("/suivi/cca/delete/{id}", name="project_cca_supprimer")
+     *
+     * @return RedirectResponse
+     */
+    public function supprimer(Cca $cca, Request $request): Response
+    {
+        $form = $this->createDeleteForm($cca);
+
+        $form->handleRequest($request);
+
+        if ($form->isValid()) {
+            $em = $this->getDoctrine()->getManager();
```

Comment:
De même qu'au dessus. Mieux d'utiliser l'injection de dépendance.

+ c'est tout nouveaux, ca vient de sortir, mais à terme `getDoctrine()` va être supprimé. https://symfony.com/blog/new-in-symfony-5-4-controller-changes

Je ferais un commit un de ces 4 pour supprimer toute les utilisations de cette méthode dans le code.